src/main/ruby/truffleruby/core/string.rb

```diff
@@ -576,7 +576,9 @@ def inspect
       end
     end
 
-    if Primitive.character_printable_p(char) && (enc == result_encoding || (ascii && char.ascii_only?))
+    if Primitive.character_printable_p(char) && unicode && char.ord < 0x7F
```

Comment:
`0x7F` comes from here: https://github.com/ruby/ruby/blob/12f7ba5ed4a07855d6a9429aa627211db3655ca7/string.c#L6049-L6050

Happy to move it to a constant.

Reply:
Do you know intuitively what it means? I guess it's maximum UTF-8 single-byte?

Reply:
https://www.utf8-chartable.de/

`< 0x7F` is everything that looks printable to an ASCII terminal

Reply:
Either constant or add a comment, yeah.

Reply:
A comment with that link seems best.
It's almost the same as `char.ascii_only?` but it also excludes `\x7F`.

Reply:
Thanks, fixed that.